job-server/src/main/scala/spark/jobserver/JobServer.scala

```diff
@@ -158,15 +160,23 @@ object JobServer {
               "context-supervisor-proxy")
 
           proxy ! ContextSupervisor.AddContextsFromConfig  // Create initial contexts
-          startWebApi(system, proxy, jobDAO, webApiPF)
+          startWebApi(system, proxy, jobDAO, webApiPF, daoActor, config)
         }
     }
   }
 
   def startWebApi(system: ActorSystem, supervisor: ActorRef, jobDAO: JobDAO,
-      webApiPF: (ActorRef, ActorRef) => WebApi) {
+      webApiPF: (ActorRef, ActorRef, HealthCheck) => WebApi, daoActor: ActorRef, config: Config) {
     val jobInfo = system.actorOf(Props(classOf[JobInfoActor], jobDAO, supervisor), "job-info")
-    webApiPF(supervisor, jobInfo).start()
+    val healthClass = Class.forName(config.getString("spark.jobserver.healthcheck"))
+    var healthCheckInst: HealthCheck = null
+    if (healthClass.getName() == "spark.jobserver.util.ActorsHealthCheck") {
```

Comment:
nit:
This instantiation could be a bit more dynamic I guess (like have a common constructor in the trait `HealthCheck` - currently it would have the arguments supervisor, jobInfo, daoActor - that every instance needs to implement). Especially if there would be multiple implementations in the future this would become a bit blown here otherwise.

But this can also be done as part of later refactorings, I see you have put a lot effort in this PR already!

Reply:
I understand the point that this might get blown in future. But having common constructor with actors as arguments may not be relevant for other implementations of healthCheck right? For ex, we have APIHealthCheck class in HealthCheck.scala. This does not take any args like supervisor, jobInfo, daoActor as they are not relevant for a simple API health check.